go/types/encode_human_readable.go

```diff
@@ -9,12 +9,73 @@ import (
 	"fmt"
 	"io"
 	"strconv"
+	"sync"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/util/writers"
 	humanize "github.com/dustin/go-humanize"
 )
 
+// Clients can register a function to return a comment that will get appended
+// to the first line of encoded structs. For example, the noms DateTime struct
+// normally gets encoded as follows:
+//    lastRefresh: DateTime {
+//      secSinceEpoch: 1.501801626877e+09,
+//    }
+//
+// By registering a commenter function that returns a nicely formatted date,
+// the struct will be coded with a comment:
+//    lastRefresh: DateTime { // 2017-08-03T16:07:06-07:00
+//      secSinceEpoch: 1.501801626877e+09,
+//    }
+
+// Function type for commenter functions
+type HRSCommentFunc func(Struct) string
+
+var (
+	commenterRegistry = map[string]map[string]HRSCommentFunc{}
+	registryLock      sync.Mutex
```

Comment:
No you'll need it as soon as multiple things want to register from separate goroutines.